### DISCUSSION TITLE
How to run single component on Doc?

### DISCUSSION THREAD

**Discussion by wildfluss** [explosion/spaCy#12065] [Help: Coding & Implementations] [answered]
I save my Doc to file like this

```
nlp = en_core_web_lg.load()
doc = nlp(text)
doc_bytes = doc.to_bytes()
```

and then I want to continue where I left off but I don't want to run the whole pipeline again, but only run my custom component on the deserialized Doc object from disk, where should I start *to do it spaCy way eg without importing my custom component implementation function but calling only spaCy APIs if possible* ?

**Comment by thomashacker** [comment, marked as answer]
Hello,

You can use `nlp.get_pipe` ([docs](https://spacy.io/api/language/#get_pipe)) do get individual components from a spaCy pipeline and use them to process `Doc` objects.

```
nlp = spacy.load("your_model")

# doc_bytes = The way you're loading in your bytes_file
doc = Doc(nlp.vocab).from_bytes(doc_bytes)

your_component = nlp.get_pipe("your_component")
result = your_component(doc)
```

You can also disable or exclude components from a pipeline when using `spacy.load()` ([docs](https://spacy.io/api/top-level#spacy.load)). 

```
spacy.load("your_model", disable=["component_to_disable"]) # Components that will be loaded but do not process any Docs
spacy.load("your_model", exclude=["component_to_exclude"]) # Components that won't be loaded into the model
spacy.load("your_model", enable=["your_component"]) # Disables every component except the components you specified
```
